src/lsp/main.ts

```diff
@@ -681,9 +809,77 @@ function getClient(timeout: number): Promise<LanguageClient> | undefined {
 
 export async function getCljFmtConfig() {
     // TODO: Figure out a reasonable timeout
-    const client = await getClient(60 * 5 * 1000);
-    const serverInfo = await getServerInfo(client);
-    return serverInfo['cljfmt-raw'];
+    const client = await getClient(60 * 5 * 1000).catch((e) => {
+        console.error(
+            `Formatting: Error waiting for clojure-lsp to start: ${e}`
+        );
+    });
+    if (client) {
+        const serverInfo = await getServerInfo(client);
+        return serverInfo['cljfmt-raw'];
+    }
+}
+
+function showMenu(
+    items: vscode.QuickPickItem[],
+    commands: Record<string, string>
+) {
+    void vscode.window
+        .showQuickPick(items, { title: 'clojure-lsp' })
+        .then((v) => {
+            if (commands[v.label]) {
+                void vscode.commands.executeCommand(commands[v.label]);
+            }
+        });
+}
+
+function stoppedMenuCommand() {
+    const START_OPTION = 'Start';
+    const START_COMMAND = 'calva.clojureLsp.start';
+    const DOWNLOAD_OPTION = 'Download configured version';
+    const DOWNLOAD_COMMAND = 'calva.clojureLsp.download';
+    const commands = {};
+    commands[START_OPTION] = START_COMMAND;
+    commands[DOWNLOAD_OPTION] = DOWNLOAD_COMMAND;
+    const items: vscode.QuickPickItem[] = [
+        {
+            label: START_OPTION,
+            description: 'Start the clojure-lsp server',
+        },
+        {
+            label: DOWNLOAD_OPTION,
+            description: `${config.getConfig().clojureLspVersion}`,
+        },
+    ];
+    showMenu(items, commands);
+}
+
+function startedMenuCommand() {
+    const STOP_OPTION = 'Stop';
+    const STOP_COMMAND = 'calva.clojureLsp.stop';
+    const INFO_OPTION = 'Show server info';
+    const INFO_COMMAND = 'calva.diagnostics.clojureLspServerInfo';
+    const LOG_OPTION = 'Open log';
+    const LOG_COMMAND = 'calva.diagnostics.openClojureLspLogFile';
+    const commands = {};
+    commands[STOP_OPTION] = STOP_COMMAND;
+    commands[INFO_OPTION] = INFO_COMMAND;
+    commands[LOG_OPTION] = LOG_COMMAND;
```

Comment:
This might just be me being picky 😄 , but you could make this just an object literal:

```typescript
{
   STOP_OPTION: STOP_COMMAND
}
```

etc.

(That key might need brackets around it - not sure.)

Reply:
The difference is that I want an object that is
```typescript
{
  'Stop': 'calva.clojureLsp.stop'
}
```

Whereas what you have there would give me:

```typescript
{
  'STOP_OPTION': 'calva.clojureLsp.stop'
}
```

Maybe you know some trick with brackets that would work? If so, I'm all for it, because the way I'm doing it now is error prone.

Reply:
Not sure why I suggested that method the first time 😄 . But brackets would work:

![image](https://user-images.githubusercontent.com/12722744/159104732-8e4da1a2-ce55-438d-90f2-9c7c3cf10a65.png)